lib/missing_step_error.ex

```diff
@@ -0,0 +1,62 @@
+defmodule MissingStepError do
+  @moduledoc """
+  Raises an error, because a feature step is missing its implementation.
+
+  The message of the error will give the user a useful code snippet where
+  variables in feature steps are converted to regex capture groups.
+  """
+
+  defexception [:message]
+
+  @number_regex ~r/(^|\s)\d+(\s|$)/
+  @single_quote_regex ~r/'[^']+'/
+  @double_quote_regex ~r/"[^"]+"/
+
+  def exception(step_text: step_text, step_type: step_type) do
```

Comment:
So typically how exceptions are defined, you use `exception/1` to turn it into a struct like `%__MODULE__{text: step_text, type: step_type}` and then define a `message/1` which turns that struct into a message string. The documentation example isn't very through, but they do describe it in the description https://hexdocs.pm/elixir/Kernel.html#defexception/1